src/Config/ConfigPanel.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// this format is came from
+// https://github.com/microsoft/vscode-extension-samples/tree/main/webview-sample
+import * as vscode from 'vscode';
+
+
+export class ConfigPanel {
+  /**
+   * Track the current panel. Only allow a single panel to exist at a time.
+   */
+  public static currentPanel: ConfigPanel|undefined;
+
+  public static readonly viewType = 'one-vscode';
+
+  private readonly _panel: vscode.WebviewPanel;
+  private readonly _extensionUri: vscode.Uri;
+  private _disposables: vscode.Disposable[] = [];
+
+  public static createOrShow(context: vscode.ExtensionContext) {
+    const column =
+        vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+
+    // If we already have a panel, show it.
+    if (ConfigPanel.currentPanel) {
+      ConfigPanel.currentPanel._panel.reveal(column);
+      ConfigPanel.currentPanel._update(context);
+      return;
+    }
+
+    // Otherwise, create a new panel.
+    const panel = vscode.window.createWebviewPanel(
+        ConfigPanel.viewType, 'ConfigurationSettings', column || vscode.ViewColumn.One, {
+          // Enable javascript in the webview
+          enableScripts: true,
+          localResourceRoots: [],
+        });
+
+    ConfigPanel.currentPanel = new ConfigPanel(panel, context);
+  }
+
+  public static kill() {
+    ConfigPanel.currentPanel ?.dispose();
+    ConfigPanel.currentPanel = undefined;
+  }
+
+  public static revive(panel: vscode.WebviewPanel, context: vscode.ExtensionContext) {
+    ConfigPanel.currentPanel = new ConfigPanel(panel, context);
+  }
+
+  private constructor(panel: vscode.WebviewPanel, context: vscode.ExtensionContext) {
+    this._panel = panel;
+    this._extensionUri = context.extensionUri;
+
+    // Set the webview's initial html content
+    this._update(context);
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+  }
+
+  public dispose() {
+    ConfigPanel.currentPanel = undefined;
+
+    // Clean up our resources
+    this._panel.dispose();
+
+    while (this._disposables.length) {
+      const x = this._disposables.pop();
+      if (x) {
+        x.dispose();
+      }
+    }
+  }
+
+  private async _update(context: vscode.ExtensionContext) {
+    const webview = this._panel.webview;
```

Comment:
This variable is not used as you removed `_getHtmlForWebview`
It would better to remove this variable too and add a TODO like `TODO implement this`

Reply:
I remove `_update` part.

TODO
- [ ] implement update logic
- [ ] implement get html from html file
is it what you mean?

Reply:
> below example is what you mean?

None of two.

You've removed the codes with my comments. But that is not my intention
- Current code itself is OK
- But I suppose you will add `_update` and related codes in next PR
- For you and colleague it would help for next to work at if you add `TODO`
- For reviewers it would help understanding the codes if you add `TODO something`

Q) Are you uncomfortable adding `TODO` comment?

Reply:
I misunderstand your comments. 
I think remove codes and add `TODO`.
Sorry, i will do that again and add `TODO` comment